src/controllers/group/index.js

```diff
@@ -50,3 +50,28 @@ const createProductIfNotGiven = async (input, models, { transaction }) => {
 
 const savePhotos = (productId, photos, { transaction }) => productPhoto
   .bulkCreate(photos.map((url) => ({ productId, url })), { transaction })
+
+export const deleteGroup = async (obj, { id }, context) => {
+  const group = await context.models.group.findByPk(id)
+
+  if (!group || await isNotDeleteable(group, context)) return 0
+
+  const productId = group.productId
+  await context.sequelize.transaction(async (t) => {
+    await context.models.groupSubscription.destroy({ where: { groupId: id } }, { transaction: t })
+    await group.destroy({}, { transaction: t })
+    await context.models.productPhoto.destroy({ where: { productId: productId } }, { transaction: t })
+    await context.models.product.destroy({ where: { id: productId } }, { transaction: t })
+  })
```

Comment:
When you have to make multiple operations on the database to produce a single change, you need to wrap in a transaction **everything**, including `SELECT` statements. You should be using the `transaction` with the `findByPk` in the first line of this function **and** also in the `isNotDeleteable` function.